services/datamanager/builtin/builtin.go

```diff
@@ -730,6 +729,21 @@ func pollFilesystem(ctx context.Context, wg *sync.WaitGroup, captureDir string,
 		case <-ctx.Done():
 			return
 		case <-t.C:
+			logger.Debug("Polling")
+			shouldDelete, err := shouldDeleteBasedOnDiskUsage(ctx, captureDir, logger)
+			if err != nil {
+				logger.Errorw("Error checking file system stats", "error", err)
+			}
+			if shouldDelete {
+				start := time.Now()
+				deletedFileCount, err := deleteFiles(ctx, syncer, captureDir, logger)
+				if err != nil {
+					duration := time.Since(start)
```

Comment:
Nice! Would ideally include the duration right after deletionFileCount and before the if-else, since useful to still include the execution time in the success case.